pkg/kubelet/cm/devicemanager/plugin/v1beta1/client.go

```diff
@@ -78,20 +87,40 @@ func (c *client) Connect() error {
 
 // Run is for running the device plugin gRPC client.
 func (c *client) Run() {
-	stream, err := c.client.ListAndWatch(context.Background(), &api.Empty{})
-	if err != nil {
-		klog.ErrorS(err, "ListAndWatch ended unexpectedly for device plugin", "resource", c.resource)
-		return
-	}
-
-	for {
-		response, err := stream.Recv()
+	var lastErr error
+	err := wait.ExponentialBackoff(c.backoff, func() (bool, error) {
+		stream, err := c.client.ListAndWatch(context.Background(), &api.Empty{})
 		if err != nil {
+			lastErr = err
 			klog.ErrorS(err, "ListAndWatch ended unexpectedly for device plugin", "resource", c.resource)
-			return
+			if err := c.Disconnect(); err != nil {
```

Comment:
It would be nice to know a real reason of failure. Is it a connection drop or something else? If it's something else, then we probably shouldn't force disconnect.